src/lib/core/overlay/overlay-directives.ts

```diff
@@ -169,11 +174,29 @@ export class ConnectedOverlayDirective implements OnDestroy {
     const originPoint = {originX: pos.originX, originY: pos.originY};
     const overlayPoint = {overlayX: pos.overlayX, overlayY: pos.overlayY};
 
-    return this._overlay.position()
+    const strategy = this._overlay.position()
       .connectedTo(this.origin.elementRef, originPoint, overlayPoint)
       .withDirection(this.dir)
       .withOffsetX(this.offsetX)
       .withOffsetY(this.offsetY);
+
+    this._handlePositionChanges(strategy);
+
+    return strategy;
+  }
+
+  private _handlePositionChanges(strategy: ConnectedPositionStrategy): void {
+    for (let i = 1; i < this.positions.length; i++) {
+      strategy.withFallbackPosition(
+          {originX: this.positions[i].originX, originY: this.positions[i].originY},
+          {overlayX: this.positions[i].overlayX, overlayY: this.positions[i].overlayY}
+      );
+    }
+
+    this._positionSubscription = strategy.onPositionChange.subscribe(
+        (pos: ConnectedOverlayPositionChange) => {
```

Comment:
This type shouldn't be necessary since it's in the generic, no? Then you could do
```ts
this._positionSubscription = 
    strategy.onPositionChange.subscribe(pos => this.positionChange.emit(pos));
```